crates/bevy_app/src/app_builder.rs

```diff
@@ -231,6 +327,30 @@ impl AppBuilder {
         self
     }
 
+    /// Init a resource to the current [App], if it does not exist yet
+    ///
+    /// Adds a resource that implements `Default` or [`FromResources`] trait.
+    /// If the resource already exists, `init_resource` does nothing.
+    ///
+    /// ## Example
+    /// ```
+    /// # use bevy_app::prelude::*;
+    /// #
+    /// struct State {
```

Comment:
State is a confusing name for this example, because of the prevalence of `StateStage` and the like.

Reply:
Agreed

Reply:
I've updated now all `State` instances in this pull req to `MyCounter`